### DISCUSSION TITLE
Integrating livewire with Spatie media library

### DISCUSSION THREAD

**Discussion by pishguy** [livewire/livewire#2479] [3. Feature ideas]
i try to found how we can integrate `Livewire` with `spatie/laravel-medialibrary` in `Laravel`. in this code i can get image information when user submit the form, but i can't save uploaded image with `spatie/laravel-medialibrary`:

```php
class ProfileComponent extends Component
{
    use WithFileUploads;

    public $photo;

//...

public function submit()
{
    $this->validate($this->rules);

    $user = User::find(auth()->user()->id);

    //...
    try {
        $user
            ->addMedia($this->photo)
            ->toMediaCollection(User::MEDIA_COLLECTION_NAME);
    } catch (FileDoesNotExist | FileIsTooBig $e) {
        dd($this->photo);
    }
}
```

or i try to use this code, but i get error too

```php
->addMediaFromDisk('/livewire-tmp/'.$this->photo->getFilename())
```

as  we know we don't have `request` method and by default `livewire` used `store` function for uploading images. how can we integrate them?

**Comment by ju5t** [comment]
We use `$this->photo->getRealPath()`. 

Also checkout my answer to the issue I ran into myself in case you're displaying a preview before `submit` is called: https://github.com/livewire/livewire/discussions/2389#discussioncomment-302690. 

In the end we did this:

```php
...
$user->addMedia($this->photo)->toMediaCollection(User::MEDIA_COLLECTION_NAME);
$this->photo = null;
...
```

And in the view a simple if statement:

```blade
@if ($photo)
temporary preview
@else
current photo                                            
@endif
```

**Reply by jomonkj** [reply]
thanks saved my time

**Reply by judgej** [reply]
Just to add to this, by passing in the side-loaded `$this->photo`, the media library will all store the original filename.

**Comment by pishguy** [comment]
@ju5t Thanks so much
